onnxruntime/core/providers/cuda/reduction/reduction_functions.cu

```diff
@@ -286,23 +285,15 @@ bool is_matrix_row_reduction(
   if (cudnn_reduce_op != CUDNN_REDUCE_TENSOR_ADD)
     return false;
 
-  // Check if all but the last axis are reduced. For example, reducing
-  // [N, C, H, W]-tensor to [W]-tensor can pass these two checks but reducing
-  // [N, C]-tensor to [N, 1]-tensor cannot.
-  if (axes.size() != rank - 1)
-    return false;
-
-  // The last reduced axis should be the second last axis. For
-  // [N, C, H, W]-input, the sorted axes should be [0, 1, 2].
-  std::sort(axes.begin(), axes.end());
-  if (axes.back() != rank - 2)
+  //empty axes, default reduction
+  if (axes.size() < 1)
```

Comment:
I guess the default reduction case can actually use this kernel.

[M,N] -> [1] can be view as [1, M*N] -> [1] ?

Reply:
I did some performance bench-marking and looks like current cudnn kernel is faster for the default reduction case.
Keeping it as is for now.

tensor shape | ORT(us) | CUDNN(us)
-- | -- | --
32, 512, 1024 | 91.225 | 46.169
1024, 4096 | 32.604 | 16.229
1024, 1024 | 6.461 | 2.824